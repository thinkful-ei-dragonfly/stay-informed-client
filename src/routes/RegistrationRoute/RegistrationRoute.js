import React, { Component } from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'

class RegistrationRoute extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegistrationSuccess = (user) => {
    const { history } = this.props
    history.push(`/dashboard`)
  }

  render() {
    return (
      <section className='container'>
        <h2>Sign up</h2>
        <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </section>
    );
  }
}

export default RegistrationRoute
