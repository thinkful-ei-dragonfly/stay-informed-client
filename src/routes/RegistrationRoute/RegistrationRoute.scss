$blue: #0063C0;
.heroAbout,
.createFormDiv,
.searchNow {
  display: flex;
}
.heroAbout {
  align-items: flex-start;
  justify-content: space-between;
  padding: 100px 0 100px 0;

  .heroAboutImg,
  .heroAboutText {
    width: 50%;
  }
  .heroAboutImg {
    text-align: right;
    padding-right: 25px;
  }
  .heroAboutText {
    padding-top: 10%;
    padding-left: 25px;

    h2 {
      max-width: 25ch;
      margin-bottom: 1em;
    }
    .submit {
      opacity: 0.75;
      transition: opacity .2s ease;
      cursor: pointer;
      &:hover {
        opacity: 1;
      }
    }
    .registerButton {
      margin-right: 50px;
      background: pink;
      color: #E9503E;
    }
    .searchButton {
      color: #0063C0;
      background: lightblue;
    }
  }
}
.createFormDiv {
  align-items: flex-start;
  justify-content: space-between;
  flex-direction: column;


  .title,
  .form-wrapper,
  .subtitle {
    text-align: left;
  }

  .title,
  .form-wrapper {
    width: 100%;
    @media (min-width: 768px) {
      width: 50%;
    }
  }
  .title {
    font-size: 1.5em;
    margin-bottom: 2em;
    @media (min-width: 768px) {
      padding-right: 25px;
      font-size: 2em;
    }
  }
  .form-wrapper {
    h3 {
      font-size: 1.25em;
    }

    @media (min-width: 768px) {
      padding: 0 25px;
      h3 {
        font-size: 1.75em;
      }
    }
  }
  @media (min-width: 768px) {
    flex-direction: row;
  }
}
