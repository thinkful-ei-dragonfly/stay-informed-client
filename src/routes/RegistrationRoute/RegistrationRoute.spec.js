import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationRoute from './RegistrationRoute'

describe(`RegistrationRoute component`, () => {
  it('renders a form.RegistrationRoute by default', () => {
    const wrapper = shallow(<RegistrationRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
