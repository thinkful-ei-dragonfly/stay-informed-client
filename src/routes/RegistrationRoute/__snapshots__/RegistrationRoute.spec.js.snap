// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationRoute component renders a form.RegistrationRoute by default 1`] = `
<section
  className="container"
>
  <h2>
    Sign up
  </h2>
  <RegistrationForm
    onRegistrationSuccess={[Function]}
  />
</section>
`;
